### PR TITLE
fix(admin-shortcode): Total goal attribute should be required field for [give_totals] #3060

### LINKED ISSUE #3060: fix(admin-shortcode): Total goal attribute should be required field for [give_totals]
## User Story
As an admin, I want Total Goal attribute of [give_totals] shortcode should be required so that I can see a positive goal and progress bar visible on the front end.

## Current Behavior
I currently can't see the progress bar and can see goal amount to be `$0` when the total_goal attribute is not provided with [give_totals] shortcode.

## Expected Behavior
I expect to have the total goal attribute to be required field for [give_totals] shortcode.

## Visuals
**Admin end - Popup of [give_totals] shortcode**
image

**Frontend - When total goal attribute is not provided in [give_totals] shortcode.**
image

**Frontend - When total goal attribute is provided in [give_totals] shortcode**
image

## Related
- https://github.com/WordImpress/Give/issues/1472

## Tasks
- [ ] Make the total goal attribute required field for [give_totals] shortcode.
- [ ] Testing

### PR DESCRIPTION
**Pull request by raftaar1191**
## Description
PR to fix #3060 

## How Has This Been Tested?
Tested by entering the shortcode by Goal amount 

## Screenshots (jpeg or gifs if applicable):
image

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/admin/shortcodes/shortcode-give-totals.php [hunk 1]
```diff
@@ -72,26 +72,29 @@ public function define_fields() {
...
-				'type'    => 'textbox',
-				'name'    => 'total_goal',
-				'label'   => __( 'Total Goal:', 'give' ),
-				'tooltip' => __( 'Enter the total goal amount that you would like to display.', 'give' ),
+				'type'     => 'textbox',
+				'name'     => 'total_goal',
+				'label'    => __( 'Total Goal:', 'give' ),
+				'tooltip'  => __( 'Enter the total goal amount that you would like to display.', 'give' ),
+				'required' => array(
+					'alert' => esc_html__( 'You must first enter the total goal amount.', 'give' ),
```
Comment (raftaar1191) [inline includes/admin/shortcodes/shortcode-give-totals.php]: @DevinWalker Need you help to update this text

**Comment by raftaar1191** [comment]
@DevinWalker do you want me to check for the value that is getting enter in the Total Goal amount too.

Kindly check the video too @DevinWalker 

Example: If Admin enter any alphabet and then click on insert shortcode button then the insert shortcode button will work as expected 


Video Link: https://screencast-o-matic.com/watch/cFhUY1b7sg

**Comment by raftaar1191** [comment]
@DevinWalker @ravinderk 


As discussed over the call. This PR is ready for merge @Sidsector9 will fix the Javascript alert message issues in https://github.com/WordImpress/Give/issues/3282

File: includes/admin/shortcodes/shortcode-give-totals.php (hunk 1 above)
Comment (ravinderk) [reply includes/admin/shortcodes/shortcode-give-totals.php]: @raftaar1191 We can add text like `Please enter a valid total goal amount`

**Pull request merged**